compiler/src/iree/compiler/Codegen/LLVMCPU/Utils.cpp

```diff
@@ -12,60 +12,89 @@
 
 #define DEBUG_TYPE "iree-llvmcpu-utils"
 
+static const char kMaxStackAllocationSizeAttrName[] =
+    "max_stack_allocation_size";
+static const char kNativeVectorSizeAttrName[] = "native_vector_size";
+
 namespace mlir::iree_compiler {
 
 static llvm::cl::opt<int32_t> clMaxAllowedNumberOfNativeVectors(
     "iree-llvmcpu-max-allowed-number-of-native-vectors",
     llvm::cl::desc("ratio used to compute the max allowed vector size"),
     llvm::cl::init(512));
 
-bool preferIntrinsicsOverAsm(IREE::HAL::ExecutableTargetAttr targetAttr) {
+std::optional<int64_t>
+getConfigMaxStackAllocationSize(DictionaryAttr targetConfig) {
+  auto attr = targetConfig.getAs<IntegerAttr>(kMaxStackAllocationSizeAttrName);
+  return attr ? std::optional<int64_t>(attr.getInt()) : std::nullopt;
```

Comment:
nit/optional: I think it's more straightforward to use control flow
```suggestion
  if (auto attr = targetConfig.getAs<IntegerAttr>(kMaxStackAllocationSizeAttrName)) {
    return attr.getInt() 
  }
  return std::nullopt;
```
Similar in the function below.